### PR TITLE
add UserControllerTest.java

### PR DESCRIPTION
**Pull request by AnastasiiaVariuchuk**

### CODE REVIEW

**Review by m-pnsk — APPROVED** [review]

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    @InjectMocks
+    private UserController userController;
+
+    @Test
+    void updateRole_validIdAndRole_ok() {
+        User.Role updatedRole = User.Role.MANAGER;
+        User updatedUser = new User();
+        updatedUser.setId(USER_ID);
+        updatedUser.setRole(updatedRole);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:39]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        when(userService.updateRole(USER_ID, updatedRole)).thenReturn(updatedUser);
+        when(userMapper.mapToDto(updatedUser)).thenReturn(new UserResponseDto());
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:42]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        UserResponseDto responseDto = userController.updateRole(USER_ID, updatedRole);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:44]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        assertNotNull(responseDto);
+    }
+
+    @Test
+    void getProfileInfo_authenticatedUser_ok() {
+        Authentication authentication = Mockito.mock(Authentication.class);
+        String userEmail = "test@example.com";
+        User user = new User();
+        user.setId(USER_ID);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:54]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        when(authentication.getName()).thenReturn(userEmail);
+        when(userService.getByEmail(userEmail)).thenReturn(Optional.of(user));
+        when(userMapper.mapToDto(user)).thenReturn(new UserResponseDto());
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:58]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        UserResponseDto responseDto = userController.getProfileInfo(authentication);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:60]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        assertNotNull(responseDto);
+    }
+
+    @Test
+    void updateProfileInfo_authenticatedUserAndUserDto_ok() {
+        Authentication authentication = Mockito.mock(Authentication.class);
+        String userEmail = "test@example.com";
+        User user = new User();
+        user.setId(USER_ID);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:70]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        UserRegistrationDto userDto = new UserRegistrationDto();
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:72]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        when(authentication.getName()).thenReturn(userEmail);
+        when(userService.getByEmail(userEmail)).thenReturn(Optional.of(user));
+        when(userMapper.mapToEntity(userDto)).thenReturn(user);
+        when(userService.update(user)).thenReturn(user);
+        when(userMapper.mapToDto(user)).thenReturn(new UserResponseDto());
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:78]:
```suggestion
```

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        UserResponseDto responseDto = userController.updateProfileInfo(authentication, userDto);
+
```
Comment (ViktorBykh) [inline src/test/java/com/example/carsharing/controller/UserControllerTest.java:80]:
```suggestion
```

**Review by ViktorBykh — APPROVED** [review]

File: src/test/java/com/example/carsharing/controller/UserControllerTest.java (hunk 10 above)
Comment (AnastasiiaVariuchuk) [reply src/test/java/com/example/carsharing/controller/UserControllerTest.java:80]: I left it to separate some logic in methods

**Pull request merged**